shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,951 @@
+            "confirmation": {"type": "boolean"},
+        }
+
+    def done_action(self):
+        return {
+            "picking_id": {"coerce": to_int, "required": True, "type": "integer"},
+            "confirmation": {"type": "boolean"},
+        }
+
+    def set_lot_confirm_action(self):
+        return {
+            "picking_id": {"coerce": to_int, "required": True, "type": "integer"},
+            "selected_line_ids": {
+                "type": "list",
+                "required": True,
+                "schema": {"coerce": to_int, "required": True, "type": "integer"},
+            },
+        }
+
+
+class ShopfloorReceptionValidatorResponse(Component):
+    _inherit = "base.shopfloor.validator.response"
+    _name = "shopfloor.reception.validator.response"
+    _usage = "reception.validator.response"
+
+    _start_state = "select_document"
+
+    # STATES
+
+    def _states(self):
+        """List of possible next states
+
+        With the schema of the data send to the client to transition
+        to the next state.
+        """
+        return {
+            "select_document": self._schema_select_document,
+            "select_line": self._schema_select_line,
+            "set_lot": self._schema_set_lot,
+            "set_quantity": self._schema_set_quantity,
+            "set_destination": self._schema_set_destination,
+            "select_dest_package": self._schema_select_dest_package,
+        }
+
+    def _start_next_states(self):
+        return {"select_document"}
+
+    def _scan_document_next_states(self):
+        return {"select_document", "select_line", "set_lot", "set_quantity"}
+
+    def _scan_line_next_states(self):
+        return {"select_line", "set_lot", "set_quantity"}
+
+    def _set_lot_next_states(self):
+        return {"select_line", "set_lot", "set_quantity"}
+
+    def _set_quantity_next_states(self):
+        return {"set_quantity", "select_line", "set_destination"}
+
+    def _set_destination_next_states(self):
+        return {"set_destination", "select_line"}
+
+    def _select_dest_package_next_states(self):
+        return {"set_lot", "select_dest_package"}
+
+    def _done_next_states(self):
+        return {"select_document", "select_line"}
+
+    def _set_lot_confirm_action_next_states(self):
+        return {"set_lot", "set_quantity"}
+
+    def _process_with_existing_pack_next_states(self):
+        return {"select_dest_package"}
+
+    def _process_with_new_pack_next_states(self):
+        return {"set_destination"}
+
+    def _process_without_pack_next_states(self):
+        return {"set_destination"}
+
+    # SCHEMAS
+
+    @property
+    def _schema_select_document(self):
+        return {
+            "pickings": self.schemas._schema_list_of(
+                self._schema_stock_picking(), required=True
+            )
+        }
+
+    def _schema_stock_picking(self, lines_with_packaging=False):
+        schema = self.schemas.picking()
```

Comment:
use picking_detail schema

Reply:
?

Reply:
It was updated to _schema_stock_picking_with_lines

Reply:
you got my point ;)